lib/KettleRequest.js

```diff
@@ -165,13 +165,22 @@ fluid.defaults("kettle.request", {
          * have handlers registered which forward to this request's success and error events.
          * The promise may be rejected or resolved by the user to represent that disposition
          * for the overall request */
-        handlerPromise: "@expand:fluid.promise()"
+        handlerPromise: "@expand:fluid.promise()",
+        /*
+         * TODO: explain what the heck outerHandlePromise is for.
+         * (It's a backward compatibility courtesy for enabling sending additional metadata with response payloads, such as status codes)
+         */
+        outerHandlerPromise: "@expand:fluid.promise()"
```

Comment:
I propose amending this to 
```
handlerPromise: "@expand:fluid.promise()",
legacyHandlerPromise: "{that}.handlerPromise",
outerHandlerPromise: "@expand:fluid.promise()"
```
and adding a TODO to deprecate outerHandlerPromise for the next breaking version of Kettle. The legacyHandlerPromise alias should help make the code connecting the promises further down a little clearer.

Reply:
Yes, I think this is reasonable